src/WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -76,34 +76,45 @@ public static void MarshalDelegateInvoke<T>(IntPtr thisPtr, Action<T> invoke)
 
         // If we are free threaded, we do not need to keep track of context.
         // This can either be if the object implements IAgileObject or the free threaded marshaler.
-        internal unsafe static bool IsFreeThreaded(IObjectReference objRef)
+        internal unsafe static bool IsFreeThreaded(IntPtr iUnknown)
         {
-            if (objRef.TryAs(InterfaceIIDs.IAgileObject_IID, out var agilePtr) >= 0)
+            if (Marshal.QueryInterface(iUnknown, ref Unsafe.AsRef(InterfaceIIDs.IAgileObject_IID), out var agilePtr) >= 0)
             {
                 Marshal.Release(agilePtr);
                 return true;
             }
-            else if (objRef.TryAs(InterfaceIIDs.IMarshal_IID, out var marshalPtr) >= 0)
+            else
             {
-                try
+                if (Marshal.QueryInterface(iUnknown, ref Unsafe.AsRef(InterfaceIIDs.IMarshal_IID), out var marshalPtr) >= 0)

```

Comment:
Nit: you can simplify this by removing the outer `else` and one indentation for this whole inner `if`, since the previous block has a `return` statement anyway, so this code will always be executed here if that branch is not taken. I mean:
```csharp
if (...)
{
    return ...;
}
else
{
    if (...)
    {
        return ...;
    }
}
```
To:
```csharp
if (...)
{
    return ...;
}

if (...)
{
    return ...;
}
```